tracing/utils.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 CloudWeGo Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracing
+
+import (
+	"time"
+
+	"github.com/cloudwego/kitex/pkg/rpcinfo"
+	"github.com/cloudwego/kitex/pkg/stats"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// recordErrorSpan log error to span
+func recordErrorSpan(span trace.Span, err error, withStackTrace bool, attributes ...attribute.KeyValue) {
+	if span == nil || err == nil {
+		return
+	}
+
+	span.SetStatus(codes.Error, err.Error())
+	span.RecordError(
+		err,
+		trace.WithAttributes(attributes...),
+		trace.WithStackTrace(withStackTrace),
```

Comment:
is it necessary to report stack? all stacks are same, because call at the same place. But Panic info is required to report, it can get by `ri.Stats().Panicked()`

Reply:
> is it necessary to report stack? all stacks are same, because call at the same place. But Panic info is required to report, it can get by `ri.Stats().Panicked()`

Makes sense, I'll change it